src/super_gradients/training/utils/media/stream.py

```diff
@@ -39,22 +50,30 @@ def run(self) -> None:
 
         Press 'q' to quit the streaming.
         """
-        while not self._stop():
-            self._display_single_frame()
+        while not self._stop() and self._display_single_frame():
+            pass
 
-    def _display_single_frame(self) -> None:
+    def _display_single_frame(self) -> bool:
         """Read a single frame from the video capture device, apply any specified frame processing,
         and display the resulting frame in the window.
 
         Also updates the FPS counter and displays it in the frame.
         """
         _ret, frame = self.cap.read()
+        if not _ret or frame is None:
+            logger.warning("Could not read frame from video capture device.")
+            return False
 
         if self.frame_processing_fn:
+            # Convert the frame to RGB since this is the format expected
+            # by the predict function
+            frame = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
```

Comment:
Why are we to assume the frame is BGR ?

Reply:
OpenCV imread and video capture always return frames in BGR color order